department-of-veterans-affairs/abd-vro
# Hoppy concurrency improvements

## What was the problem?
Hoppy users had to poll the client's readiness status when starting and stopping the client. Additionally, this fixes a bug where the caller's event loop was closed by hoppy when no event loop was specified on initialization.

Associated tickets or Slack threads:
- #2149 
- #2150 
- #2151 

## How does this fix it?[^1]

Adds `async def` signature to client start/stop functions, and checks if a custom event loop was specified on initialization before closing it.

## How to test this PR
- run `pytest` in `shared/lib-hoppy`


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
